.github/workflows/test.yml

```diff
@@ -1,59 +1,34 @@
----
-    name: Test
-    on:
-      push:
-        branches:
-          - main
-      pull_request:
-    
-    jobs:
-      test:
-        strategy:
-          matrix:
-            go-version: [ 1.21.x ]
-            goarch: [ "amd64" ]
-        runs-on: ubuntu-latest
-        steps:
-        - name: Checkout code
-          uses: actions/checkout@v3
-    
-        - name: Install Go
-          uses: actions/setup-go@v3
-          with:
-            go-version: ${{ matrix.go-version }}
-          env:
-            GOARCH: ${{ matrix.goarch }}
-    
-        - name: Test
-          run: make test
-          working-directory: ./
+name: Test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
 
+jobs:
+  test:
+    strategy:
+      matrix:
+        go-version: [1.21.x]
+        goarch: ["amd64"]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
 
-        - name: Archive code coverage results
-          uses: actions/upload-artifact@v3
-          with:
-            name: code-coverage-report
-            path: coverage.out
+      - name: Install Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: ${{ matrix.go-version }}
+        env:
+          GOARCH: ${{ matrix.goarch }}
 
-      sonar-cloud:
-        needs: test
-        name: SonarCloud
-        runs-on: ubuntu-latest
-        steps:
-          - name: Checkout Code
-            uses: actions/checkout@v3
-            with:
-              submodules: recursive
-              fetch-depth: 0 # Shallow clones should be disabled for a better relevancy of analysis
+      - name: Test
+        run: make test
+        working-directory: ./
 
-          - name: Download code coverage results
-            uses: actions/download-artifact@v3
-            with:
-              name: code-coverage-report
-
-          - name: SonarCloud analysis
-            uses: SonarSource/sonarcloud-github-action@master
-            env:
-              GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-              SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
-              SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+      - name: Archive code coverage results
+        uses: actions/upload-artifact@v4
+        with:
+          name: code-coverage-report
+          path: coverage.out
```

Comment:
If we are removing the sonar-cloud analysis, it safe to remove this part as well.


```suggestion
```

Reply:
Added sonar-cloud